File: 프로그래머스/unrated/150370. 개인정보 수집 유효기간/개인정보 수집 유효기간.py
def solution(today, terms, privacies):
    answer = []
    term_dic = {}
    for term in terms:
        type, month = term.split()
        term_dic[type] = int(month)
        
    for i in range(len(privacies)):
        date, type = privacies[i].split()
        year, month, day = map(int, date.split("."))
        month += term_dic[type]
        year += month // 12
        if month % 12 == 0:
            year -= 1
            month = 12
        else:
            month = month % 12
        due_date = ".".join([str(year), ("0"+str(month))[-2:], ("0"+str(day))[-2:]])
        if today >= due_date: answer.append(i+1)
    return answer

File: 프로그래머스/unrated/150370. 개인정보 수집 유효기간/개인정보 수집 유효기간.java
import java.util.*;

class Solution {
    int convert(String date) {
        String[] temp = date.split("[.]");
        int year = Integer.parseInt(temp[0].substring(2, 4));
        int month = Integer.parseInt(temp[1]);
        int day = Integer.parseInt(temp[2]);
        
        return year*12*30+month*30+day;
    }
    public int[] solution(String today, String[] terms, String[] privacies) {
        List<Integer> answer = new ArrayList<>();
        Map<String, Integer> map = new HashMap<>();
        
        for (String term: terms) {
            String[] temp = term.split(" ");
            map.put(temp[0], Integer.parseInt(temp[1])*30);
        }
        
        int todayConverted = convert(today);
        for (int i=0;i<privacies.length;i++) {
            String[] temp = privacies[i].split(" ");
            int destroy = convert(temp[0]) + map.get(temp[1]);
            if (todayConverted >= destroy) answer.add(i+1);
        }
        
        return answer.stream().mapToInt(integer -> integer).toArray();
    }
}